### PR TITLE
__put_name_here__

### PR DESCRIPTION
**Pull request by vovapi**
## What's done:
* Remove callback method of processing async fetched data.
* Add config file (default location is `configs/main.yml`
* Separate fetch and update functions
* Some refactoring here and there
* Add support for `/token_price` endpoint
## TODO
* Proper logging
* Fetched data caching

### CODE REVIEW

File: models/google_sheets_api.py [hunk 1]
```diff
@@ -61,6 +64,7 @@ def _get_credentials(self):
         store = Storage(credential_path)
         credentials = store.get()
         if not credentials or credentials.invalid:
+            flow = client.flow
```
Comment (bashalex) [inline models/google_sheets_api.py]: ??

File: models/coinmarketcap.py [hunk 2]
```diff
@@ -26,15 +26,25 @@ def __repr__(self):
 class CoinmarketcapAPI(Fetcher):
     _COINMARKETCAP_URL = 'https://api.coinmarketcap.com/v1/ticker?limit={}'
 
+    # Coins that doesnt have
```
Comment (bashalex) [inline models/coinmarketcap.py]: Coins that do not have what?

File: README.md [hunk 3]
```diff
@@ -2,18 +2,23 @@
 
 Requires authorisation via etherionlab@gmail.com
 
+## Config
+See [sample.yml](configs/sample.yml)
+
```
Comment (bashalex) [inline README.md]: I would add that by default `configs/main.yml` fail is used.

File: models/google_sheets_api.py [hunk 4]
```diff
@@ -29,14 +29,17 @@ class SheetsAPI:
...
 
-    def __init__(self, sheets_id='1kFi2uGX3RZYFj76cXNE93ys3hJY49XhHtPCj_Sc5nNE', app_name='Assets Parser'):
+    def __init__(self, sheets_id, app_name='Assets Parser', secret_file=None):
+
         self._SHEETS_ID = sheets_id
+        if secret_file is not None:
+            self._CLIENT_SECRET_FILE = secret_file
         self._APPLICATION_NAME = app_name
         self._credentials = self._get_credentials()
         http = self._credentials.authorize(httplib2.Http())
```
Comment (bashalex) [inline models/google_sheets_api.py:45]: Let's add `cache_discovery=False` to `discovery.build()` to get rid of warning messages as suggested here: https://stackoverflow.com/a/44518587/4793743

File: update_table.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
+import asyncio
```
Comment (bashalex) [inline update_table.py]: redundant dependency

File: update_table.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        portfolio_value.get('ETH') / number_of_tokens / initial_token_eth_price - 1,  # ETH price lifetime change
+    ]
+    api.add_daily_performance_row(row)
+
+    return row
+
+
+def update_table(config):
+    api = SheetsAPI(sheets_id=config['sheets']['id'], secret_file=config['sheets']['secret_file'])
+    date = time.strftime('%Y-%m-%d %H:%M %Z')
```
Comment (bashalex) [inline update_table.py]: I think would be better to convert datetime to UTC and print it without timezone

**Review by bashalex — CHANGES_REQUESTED** [review]

File: models/google_sheets_api.py (hunk 4 above)
Comment (vovapi) [reply models/google_sheets_api.py:45]: @bashalex done in 08067ff

File: models/google_sheets_api.py (hunk 1 above)
Comment (vovapi) [reply models/google_sheets_api.py]: @bashalex fixed in 08067ff

File: models/coinmarketcap.py (hunk 2 above)
Comment (vovapi) [reply models/coinmarketcap.py]: @bashalex this used to be a list for coins that does not have proper info (e.g. `{"update_time": none}`). I've moved from this, but it is somehow got here.
Fixed in 9871d9d

File: update_table.py (hunk 6 above)
Comment (vovapi) [reply update_table.py]: @bashalex Done in cf85346

File: update_table.py (hunk 5 above)
Comment (vovapi) [reply update_table.py]: cf85346

**Review by bashalex — APPROVED** [review]

**Pull request merged**